experiments/conductor/cmd/runner/crd_generator_commands.go

```diff
@@ -0,0 +1,320 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT="$(git rev-parse --show-toplevel)"
+cd ${REPO_ROOT}/dev/tools/controllerbuilder
+
+go run . generate-types \
+    --service <PROTO_PACKAGE> \
+    --api-version <CRD_GROUP>/<CRD_VERSION> \
+    --resource <CRD_KIND>:<PROTO_RESOURCE>
+
+go run . generate-mapper \
+    --service <PROTO_PACKAGE> \
+    --api-version <CRD_GROUP>/<CRD_VERSION>
+
+cd ${REPO_ROOT}
+dev/tasks/generate-crds
+
+go run -mod=readonly golang.org/x/tools/cmd/goimports@latest -w pkg/controller/direct/<SERVICE>/
+`
+
+// NOT USED
+const UPDATE_GENERATE_SCRIPT_PROMPT = `
+Please update the apis/<SERVICE>/v1alpha1/generate.sh script for the <SERVICE> API to generate the CRD for the <CRD_KIND> resource.
+
+The generate.sh script is located at apis/<SERVICE>/v1alpha1/generate.sh.
+
+Add the parameter <TICK> --resource <CRD_KIND>:<PROTO_RESOURCE><TICK> to the <TICK>go run . generate-types --api-version <CRD_GROUP>/<CRD_VERSION>  <TICK> command.
+
+At the end of the script, ensure the following lines are present:
+
+cd ${REPO_ROOT}
+dev/tasks/generate-crds
+
+go run -mod=readonly golang.org/x/tools/cmd/goimports@latest -w pkg/controller/direct/<SERVICE>/
+`
+
+// NOT USED
+func generateCRDFromScripts(opts *RunnerOptions, branch Branch) {
+	close := setLoggingWriter(opts, branch)
+	defer close()
+	workDir := filepath.Join(opts.branchRepoDir)
+
+	var out strings.Builder
+	checkoutBranch(branch, workDir, &out)
+
+	// Create the apis/<service>/<version> directory
+	serviceDir := filepath.Join(workDir, "apis", branch.Group, "v1alpha1")
+	if err := os.MkdirAll(serviceDir, 0755); err != nil {
+		log.Fatal(err)
+	}
+
+	// Create or update generate.sh
+	scriptPath := filepath.Join(serviceDir, "generate.sh")
+	// Check if generate.sh already exists.
+	if _, err := os.Stat(scriptPath); errors.Is(err, os.ErrNotExist) {
+		// File doesn't exist, use template approach
+		log.Printf("Creating new generate.sh at %s", scriptPath)
+
+		// Replace template markers with actual values and write to file
+		writeTemplateToFile(branch, scriptPath, GENERATOR_SCRIPT_TEMPLATE)
+	} else {
+		// File exists, use codebot to update it
+		log.Printf("Updating existing generate.sh at %s", scriptPath)
+
+		// Delete then write the prompt file.
+		promptPath := filepath.Join(workDir, "mockgcp", "crdgen_prompt.txt")
+		writeTemplateToFile(branch, promptPath, UPDATE_GENERATE_SCRIPT_PROMPT)
+
+		cfg := CommandConfig{
+			Name:    "CODEBOT GENERATE",
+			Cmd:     "codebot",
+			Args:    []string{"--ui-type=prompt", "--prompt=mockgcp/crdgen_prompt.txt"},
+			WorkDir: workDir,
+		}
+		if err := executeCommand(cfg, &out); err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	// Add and commit changes
+	scriptRelativePath := fmt.Sprintf("apis/%s/v1alpha1/generate.sh", branch.Group)
```

Comment:
paths should be contructed using filepath.Join, at least for the slashes.

Reply:
of course. will change.